### PR TITLE
Fix unconsistency between code and doc

### PR DESCRIPTION
**Pull request by PeiweiHu**
Update old documentation. I already checked them manually. 

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/EVP_RAND.pod [hunk 1]
```diff
@@ -62,7 +62,7 @@ EVP_RAND_STATE_ERROR - EVP RAND routines
                      const unsigned char *ent, size_t ent_len,
                      const unsigned char *addin, size_t addin_len);
  int EVP_RAND_nonce(EVP_RAND_CTX *ctx, unsigned char *out, size_t outlen);
- int EVP_RAND_enable_locking(EVP_RAND_CTX *ctx);
+ int EVP_RAND_enable_locking(EVP_RAND_CTX *rand);
```
Comment (paulidale) [inline doc/man3/EVP_RAND.pod:65]: Why change from _ctx_ to _rand_ for just this one call?
The remaining calls still use _ctx_.

File: doc/man3/RSA_set_method.pod [hunk 2]
```diff
@@ -16,15 +16,15 @@ L<openssl_user_macros(7)>:
...
  int RSA_set_method(RSA *rsa, const RSA_METHOD *meth);
 
- RSA_METHOD *RSA_get_method(const RSA *rsa);
+ const RSA_METHOD *RSA_get_method(const RSA *rsa);
 
- RSA_METHOD *RSA_PKCS1_OpenSSL(void);
+ const RSA_METHOD *RSA_PKCS1_OpenSSL(void);
 
- int RSA_flags(const RSA *rsa);
+ int RSA_flags(const RSA *r);
```
Comment (paulidale) [inline doc/man3/RSA_set_method.pod:27]: This should have stayed _rsa_.

**Review by paulidale — COMMENTED** [review]
I'm not sure that some of these changes are beneficial.
The doc should be as readable as possible.  Not matching the code isn't a big issue IMO.

File: doc/man3/EVP_RAND.pod (hunk 1 above)
Comment (PeiweiHu) [reply doc/man3/EVP_RAND.pod:65]: All the changes I made are based on their defination in source code. For example, this one is defined as **int EVP_RAND_enable_locking(EVP_RAND_CTX *rand)** in crypto/evp/evp_rand.c. IMO if ctx is more readable than rand, source code is better to keep ctx too.

Comment (PeiweiHu) [reply doc/man3/EVP_RAND.pod:65]: That's fine if you think it's not a deal. I will close this pr and only keep the change like miss **const** keyword and etc.

Comment (PeiweiHu) [reply doc/man3/EVP_RAND.pod:65]: That's fine if you think it's not a deal. I will close this pr and only keep the change like miss **const** keyword and etc.

File: doc/man3/RSA_set_method.pod (hunk 2 above)
Comment (PeiweiHu) [reply doc/man3/RSA_set_method.pod:27]: defined as int RSA_flags(const RSA *r) in crypto/rsa/rsa_crpt.c

**Pull request closed without merging**